ocaml/xapi/xapi_pool_update.ml

```diff
@@ -445,7 +445,16 @@ let resync_host ~__context ~host =
       ) update_refs;
     Create_misc.create_updates_requiring_reboot_info ~__context ~host
   end
-  else Db.Host.set_updates ~__context ~self:host ~value:[]
+  else Db.Host.set_updates ~__context ~self:host ~value:[];
+
+  (* Remove any pool_patch objects that don't have a corresponding pool_update object *)
+  let pool_patchs = Db.Pool_patch.get_all ~__context in
```

Comment:
This should be `pool_patches` and likewise below. Removing unused patches can be expressed more concisely with `List.filter`:


    Db.Pool_patch.get_all ~__context
    |> List.map (fun self -> Db.Pool_patch.get_pool_update ~__context ~self)
    |> List.filter Ref.null
    |> List.iter (fun self -> Db.Pool_patch.destroy ~__context ~self)